### PR TITLE
Private setup methods for AndroidDevice.

### PR DESCRIPTION
**Pull request by iansf**

### CODE REVIEW

File: lib/src/device.dart [hunk 1]
```diff
@@ -41,14 +48,30 @@ abstract class _Device {
 }
 
 class AndroidDevice extends _Device {
+  static const String _ADB_PATH = 'adb';
+
   static const String className = 'AndroidDevice';
   static final String defaultDeviceID = 'default';
 
+  static final Logger logging = new Logger(className);
```
Comment (devoncarew) [inline lib/src/device.dart]: I tend to create loggers at the library level - not the class - but am not sure that's a best practice :)

File: lib/src/process_wrapper.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library sky_tools.device;
+
+import 'dart:io';
+import 'package:logging/logging.dart';
+
+final Logger logging = new Logger('');
+String runSyncChecked(List<String> cmd) {
```
Comment (devoncarew) [inline lib/src/process_wrapper.dart]: `runSyncChecked` ==> `runCheckedSync`? That would follow some other `Sync` / async method names in the code lib.

File: lib/src/process_wrapper.dart [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+library sky_tools.device;
```
Comment (devoncarew) [inline lib/src/process_wrapper.dart]: `device` ==> `process_wrapper`

File: lib/src/process_wrapper.dart [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import 'dart:io';
+import 'package:logging/logging.dart';
+
+final Logger logging = new Logger('');
```
Comment (devoncarew) [inline lib/src/process_wrapper.dart]: Private and use the lib name as the string? `_logger = new Logger('sky_tools.process_wrapper')`

File: lib/src/device.dart [hunk 5]
```diff
@@ -64,4 +87,103 @@ class AndroidDevice extends _Device {
...
+        return false;
+      }
+      if (sdkVersionParsed < 22) {
+        logging.severe('Version "$sdkVersion" of the Android SDK is too old. '
+            'Please install Lollipop (version 22) or later.');
+        return false;
+      }
+      return true;
+    } catch (e, stack) {
+      logging.severe('Unexpected failure from adb: ');
```
Comment (devoncarew) [inline lib/src/device.dart]: `..., e, st);`, instead of logging on 3 separate lines

File: lib/src/device.dart (hunk 1 above)
Comment (iansf) [inline lib/src/device.dart]: Good idea

File: lib/src/process_wrapper.dart (hunk 2 above)
Comment (iansf) [inline lib/src/process_wrapper.dart]: Sounds good

File: lib/src/process_wrapper.dart (hunk 3 above)
Comment (iansf) [inline lib/src/process_wrapper.dart]: Oops :-)

File: lib/src/process_wrapper.dart (hunk 4 above)
Comment (iansf) [inline lib/src/process_wrapper.dart]: Done

File: lib/src/device.dart (hunk 5 above)
Comment (iansf) [inline lib/src/device.dart]: Done

**Comment by devoncarew** [comment]
:+1: lgtm

**Pull request merged**